text/0000-export-to-oci.md

```diff
@@ -0,0 +1,213 @@
+  - It will update the report.toml file with all the tags and require information to verify the image once it is pushed into a registry
+- OTHERWISE it will behave as it is right now
+
+#### `report.toml` (TOML)
+
+The new information to be  added into the `report.toml` file can be summarize as follows:
+
+```toml
+[export]
+[[export.oci]]
+digest = "<image digest>"
+manifest-size = "<manifest size in bytes>"
+compression-algorithm = "<used by gcr>"
+library-url = "https://github.com/google/go-containerregistry/"
+library-language="go"
+```
+Where:
+- **If** the app image was exported using the `-layer` flag, the export section will be added to the report
+  - `digest` MUST contain the image digest calculated based on compressed layers
+  - `manifest-size` MUST contain the manifest size in bytes
+  - `compression-algorithm` MUST contain the information of the algorithm used by GCR to compute the digest
+  - `library-url` MUST contain the url of the library used to export the image
+  - `library-language` MUST contain the client's library programming language used to export the image
+
+The main idea of this new section in the `report.toml` file is to provide the information of the **expected** digest of the image if it is exported to a registry using a particular implementation. This information can be used by other tools (like publish) to complete the operation and verify the consistency of the image.
+
+# Migration
+[migration]: #migration
+
+<!--
+This section should document breaks to public API and breaks in compatibility due to this RFC's proposed changes. In addition, it should document the proposed steps that one would need to take to work through these changes. Care should be give to include all applicable personas, such as platform developers, buildpack developers, buildpack users and consumers of buildpack images.
+-->
+# Drawbacks
+[drawbacks]: #drawbacks
+
+- We could increase the disk space if we not managed the duplication of saving the layers on disk. Currently the Cache implementation (used when daemon is ON) saved the layers tarballs on disk, because the current proposal is exporting the whole image on disk it will also require more space to save the layers for the OCI format in the `blobs` folder.
+
+# Alternatives
+[alternatives]: #alternatives
+
+## Redesign the current Cache
+
+Another potential solution could be to export the OCI image along with the current Cache concept and expose some contract for `Platform` to interact with this Cache and extract the final OCI image. At high level, the solution can be represented with the following container diagram from C4 model
+
+![](https://i.imgur.com/xl4gL1G.png)
+
+Notice that on this solution, because the Cache is an internal component from the Lifecycle implementation we will have to expose some kind of specification for `Platforms` to understand its format and been able to read the OCI image.
```

Comment:
```suggestion
Notice that on this solution, because the Launch Cache is an internal component from the Lifecycle implementation we will have to expose some kind of specification for `Platforms` to understand its format and been able to read the OCI image.
```